codemirror.ts

```diff
@@ -0,0 +1,149 @@
+import { EditorView, minimalSetup } from "codemirror";
+import { linter, Diagnostic } from "@codemirror/lint";
+import { closeBrackets } from "@codemirror/autocomplete";
+import { StreamLanguage } from "@codemirror/language";
+import { tags } from "@lezer/highlight"
+import { HighlightStyle, syntaxHighlighting } from "@codemirror/language"
+import init, { lint } from "./pkg/pest_site.js";
+
+const editorDom = document.querySelector<HTMLDivElement>(".editor")!;
+const gridDom = document.querySelector<HTMLDivElement>(".editor-grid")!;
+const inputDom = document.querySelector<HTMLDivElement>(".editor-input")!;
+const outputDom =
+  document.querySelector<HTMLTextAreaElement>(".editor-output")!;
+const modeBtn = document.querySelector<HTMLButtonElement>("#modeBtn")!;
+
+// let windowHeight = window.innerHeight;
+// let current_data = null;
+// CodeMirror.defineSimpleMode("pest", {
+//   start: [
+//     { regex: /"/, token: "string", next: "string" },
+//     {
+//       regex: /'(?:[^'\\]|\\(?:[nrt0'"]|x[\da-fA-F]{2}|u\{[\da-fA-F]{6}\}))'/,
+//       token: "string",
+//     },
+//     { regex: /\/\/.*/, token: "comment" },
+//     { regex: /\d+/, token: "number" },
+//     { regex: /[~|*+?&!]|(\.\.)/, token: "operator" },
+//     { regex: /[a-zA-Z_]\w*/, token: "variable" },
+//     { regex: /=/, next: "mod" },
+//   ],
+//   mod: [
+//     { regex: /\{/, next: "start" },
+//     { regex: /[_@!$]/, token: "operator-2" },
+//   ],
+//   string: [
+//     { regex: /"/, token: "string", next: "start" },
+//     { regex: /(?:[^\\"]|\\(?:.|$))*/, token: "string" },
+//   ],
+//   meta: {
+//     dontIndentStates: ["comment"],
+//     lineComment: "//",
+//   },
+// });
```

Comment:
It can be deleted -- I kept it because I was (and still am) having trouble with codemirror v6's language mode, but it seems better to just use the older file for that.